### PR TITLE
[release/6.0-maui] Add back HAVE_SYS_ZLIB support

### PR DESCRIPTION
**Pull request by lambdageek**
In particular this re-enables embedded PDB support on iOS and Android

The corresponding PR for `main` is #63365

### CODE REVIEW

File: src/mono/cmake/configure.cmake [hunk 1]
```diff
@@ -79,6 +80,21 @@ ac_check_funcs (
...
+check_c_source_compiles(
+  "
+  #include <zlib.h>
+  int main(void)
+  {
+    #if defined(ZLIB_VERNUM) && (ZLIB_VERNUM >= 0x1230)
+    #else
+    #error No good zlib found
+    #endif
+    return 0;
```
Comment (lambdageek) [inline src/mono/cmake/configure.cmake:92]: This is copied from

https://github.com/mono/mono/blob/a791978a6e4312698137be451c648b650a8c8663/configure.ac#L728-L732

but I'm not sure if that's necessary.

**Comment by thaystg** [comment]
LGTM, the embedded pdb is now working on ios.

**Comment by directhex** [comment]
If memory serves, this stuff was disabled because in some scenarios it was inexplicably broken. So if it's legit working, then fine, seal of approval. But it really needs every viable permutation double-checked

**Comment by thaystg** [comment]
As far as I remember, what was inexplicably broken, was fixed here: https://github.com/mono/mono/pull/18895

**Comment by directhex** [comment]
Awesome, thanks Thays

**Comment by lambdageek** [comment]
I don't understand this `runtime (Build Browser wasm Release AllSubsets_Mono_WasmBuildTests)` failure:

```
src/libraries/workloads-testing.targets(105,5): error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "InstallWorkloadFromArtifacts" task failed unexpectedly.
System.IO.DirectoryNotFoundException: Could not find a part of the path '/__w/1/s/artifacts/bin/dotnet-workload/sdk-manifests/6.0.200'.
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserDirectories(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.EnumerateDirectories(String path, String searchPattern, EnumerationOptions enumerationOptions)
   at Microsoft.Workload.Build.Tasks.InstallWorkloadFromArtifacts.FindSubDirIgnoringCase(String parentDir, String dirName) in /_/src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs:line 168
   at Microsoft.Workload.Build.Tasks.InstallWorkloadFromArtifacts.InstallWorkloadManifest(String name, String version, String nugetConfigContents, Boolean stopOnMissing) in /_/src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs:line 101
   at Microsoft.Workload.Build.Tasks.InstallWorkloadFromArtifacts.Execute() in /_/src/tasks/WorkloadBuildTasks/InstallWorkloadFromArtifacts.cs:line 58
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
```

**Comment by steveisok** [comment]
> I don't understand this `runtime (Build Browser wasm Release AllSubsets_Mono_WasmBuildTests)` failure:

Likely caused by https://github.com/dotnet/runtime/pull/62787.  I thought the failures were unrelated and missed that one.  I'll look into it.

**Review by steveisok — APPROVED** [review]

**Comment by lewing** [comment]
@lambdageek should this go to release/6.0 as well?

**Comment by lambdageek** [comment]
> @lambdageek should this go to release/6.0 as well?

@lewing I don't see a lot of motivation for it.  Note that embedded PDB support for WASM was always enabled (it was already part of the ifdef).

**Comment by steveisok** [comment]
> > @lambdageek should this go to release/6.0 as well?
> 
> 
> 
> @lewing I don't see a lot of motivation for it.  Note that embedded PDB support for WASM was always enabled (it was already part of the ifdef).

It'll make it to release/6.0 when we merge it with the Maui branch commits.

**Pull request merged**